org.eclipse.richbeans.widgets/src/org/eclipse/richbeans/widgets/scalebox/NumberBox.java

```diff
@@ -98,7 +98,7 @@
 	// FIXME Widgets should not have a reference to the controller.
 	// This breaks the model/view/controller design because the controller
 	// is referenced by the view to look up field values.
-	private IBeanController beanService;
+	private IBeanController<Object> beanService;
```

Comment:
There is no code in GDA that uses this IBeanController outside this class so yes it should be removed. Will improve encapsulation and make it a bit more MVCish.